src/Sylius/Component/Taxation/Model/TaxRate.php

```diff
@@ -120,4 +126,48 @@ public function getLabel(): ?string
     {
         return sprintf('%s (%s%%)', $this->name, $this->getAmountAsPercentage());
     }
+
+    public function getStartDate(): ?DateTimeInterface
+    {
+        return $this->startDate;
+    }
+
+    public function setStartDate(?DateTimeInterface $startDate): void
+    {
+        $this->startDate = $startDate;
+    }
+
+    public function getEndDate(): ?DateTimeInterface
+    {
+        return $this->endDate;
+    }
+
+    /**
+     * @throws Exception
+     */
+    public function setEndDate(?DateTimeInterface $endDate): void
+    {
+        if ($this->startDate != null && $endDate < $this->startDate) {
+            throw new Exception("The tax rate should not end before it starts");
+        }
+
+        $this->endDate = $endDate;
+    }
+
+    public function isInDate(DateTimeInterface $date): bool
+    {
+        if (null === $this->startDate && null === $this->endDate) {
+            return true;
+        }
+
+        if (null === $this->startDate) {
+            return $date <= $this->endDate;
+        }
+
+        if (null === $this->endDate) {
+            return $date >= $this->startDate;
+        }
+
+        return $this->startDate <= $date && $this->endDate >= $date;
```

Comment:
```suggestion
        return $this->startDate <= $date && $this->endDate >= $date;
```

I belive it will work as well, but maybe I'm wrong.

Reply:
It has to be the way it is

Reply:
it has been changed to shorter version